### ISSUE TITLE
Create devcontainer from existing docker-compose throws --no-same-owner error

### ISSUE THREAD

**Issue by danielfiniki** [microsoft/vscode-remote-release#2319]
I tried to open a existing (but new, so pretty much bare) docker-compose file to give the remote dev plugins a go. If I do a regular compose up it works like I'd expect it to. I can even create a devcontainer from the Dockerfile, but it doesn't work from the compose file.
I get this error:

```
[2020-02-08T16:42:44.403Z] [PID 3216] [8535 ms] Start: Run in container: mkdir -p /root/.vscode-server/bin/ae08d5460b5a45169385ff3fd44208f431992451_1581180164403
[2020-02-08T16:42:44.417Z] [PID 3216] [8549 ms] 
[2020-02-08T16:42:44.418Z] [PID 3216] [8550 ms] 
[2020-02-08T16:42:44.420Z] [PID 3216] [8552 ms] Start: Run: docker exec -w /root/.vscode-server/bin/ae08d5460b5a45169385ff3fd44208f431992451_1581180164403 -u root -i 93af101b03b32d71b5b4f865579c88cd4de0d401cd2e46793be3bf5099794f8d tar --no-same-owner -xz --strip-components 1


[2020-02-08T16:42:44.905Z] [PID 3216] [9037 ms] Command failed: C:\Program Files\Docker\Docker\resources\bin\docker.exe exec -w /root/.vscode-server/bin/ae08d5460b5a45169385ff3fd44208f431992451_1581180164403 -u root -i 93af101b03b32d71b5b4f865579c88cd4de0d401cd2e46793be3bf5099794f8d tar --no-same-owner -xz --strip-components 1
```

Someone [here](https://github.com/microsoft/vscode-remote-release/issues/28) described a similar issue, but I tried everything I could come up with in that department. I tried to [this](https://code.visualstudio.com/docs/remote/containers-advanced#_adding-a-nonroot-user-to-your-dev-container), so adding a user in the Dockerfile, and setting the user accordingly in the compose and the devcontainer.json. Still no dice with the compose file.
The weird thing is creating it from the Dockerfile itself works fine.

This is the Dockerfile_dev:
```
FROM tiangolo/uwsgi-nginx:python3.7
ENV PYTHONUNBUFFERED=1

RUN apt-get update && \
    apt-get install -y nano
WORKDIR /src
COPY /requirements.txt /requirements_dev.txt /src/
RUN pip install -r requirements_dev.txt
COPY ./ /src/

EXPOSE 8000
STOPSIGNAL SIGINT
ENTRYPOINT ["python", "manage.py"]
CMD ["runserver", "0.0.0.0:8000"]
```
And this is the docker-compose_dev.yml:
```
version: '3.7'

services:
  web:
    container_name: web
    build:
      context: .
      dockerfile: Dockerfile_dev
    volumes:
      - ./:/src
      - ./db.sqlite3:/db/db.sqlite3
    ports:
      - 8002:8000
```
The Django part is basically just the default stuff.

- VSCode Version: 1.42.0
- Local OS Version: Windows 10 10.0.18363
- Remote OS Version: tiangolo/uwsgi-nginx:python3.7
- Remote Extension/Connection Type: Docker
- Docker Engine: v19.03.5
- Remote-Containers extension version: 0.101.0

Steps to Reproduce:

1. Create a default Django project
2. Try to "Reopen in Container" for the docker-compose_dev.yml file I supplied - this should give you the error from above
3. Try to "Reopen in Container" for the Dockerfile_dev by itself - this should work

Maybe I'm just missing something, and this is not even a bug,  but I have spent all day trying to figure out why it's not working with docker-compose.

**Comment by chrmarti** [comment]
@danielfiniki Could you attach the full log? Does it work when you run `docker-compose up` yourself and then use `F1` > `Remote-Containers: Attach to Running Container...`?

**Comment by danielfiniki** [comment]
@chrmarti Sure, here's the full log:
```
[2020-02-18T22:14:29.775Z] [PID 3960] [3 ms] Setting up container for folder or workspace: c:\Git\pawn


[2020-02-18T22:14:29.773Z] [PID 3960] [1 ms] Start: Resolving remote
[2020-02-18T22:14:29.801Z] [PID 3960] [29 ms] Start: Check Docker is running
[2020-02-18T22:14:31.079Z] [PID 3960] [1307 ms] Start: Inspecting container
[2020-02-18T22:14:31.115Z] [PID 3960] [1343 ms] Start: Run in container: uname -m
[2020-02-18T22:14:31.445Z] [PID 3960] [1673 ms] x86_64
[2020-02-18T22:14:31.446Z] [PID 3960] [1674 ms] 
[2020-02-18T22:14:31.446Z] [PID 3960] [1674 ms] Start: Run in container: (cat /etc/os-release || cat /usr/lib/os-release) 2>/dev/null
[2020-02-18T22:14:31.451Z] [PID 3960] [1679 ms] PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
VERSION_CODENAME=stretch
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
[2020-02-18T22:14:31.452Z] [PID 3960] [1680 ms] 
[2020-02-18T22:14:31.452Z] [PID 3960] [1680 ms] Start: Run in container: cat /etc/passwd
[2020-02-18T22:14:31.458Z] [PID 3960] [1686 ms] Start: Setup shutdown monitor
[2020-02-18T22:14:31.463Z] [PID 3960] [1691 ms] Forking shutdown monitor: c:\Users\danie\.vscode\extensions\ms-vscode-remote.remote-containers-0.101.0\dist\shutdownMonitorProcess \\.\pipe\vscode-remote-containers-62e448e8d5d900fbf686b8eeed10c577c184fdfb-sock dockerCompose Info c:\Users\danie\AppData\Roaming\Code\logs\20200218T230738\exthost1\ms-vscode-remote.remote-containers
[2020-02-18T22:14:31.473Z] [PID 3960] [1701 ms] Start: Run in container: test -d /root/.vscode-server
[2020-02-18T22:14:31.491Z] [PID 3960] [1719 ms] 
[2020-02-18T22:14:31.491Z] [PID 3960] [1719 ms] 
[2020-02-18T22:14:31.491Z] [PID 3960] [1719 ms] Start: Run in container: set -o noclobber ; mkdir -p '/root/.vscode-server/data/Machine' && { > '/root/.vscode-server/data/Machine/.writeMachineSettingsMarker' ; } 2> /dev/null
[2020-02-18T22:14:31.498Z] [PID 3960] [1726 ms] 
[2020-02-18T22:14:31.498Z] [PID 3960] [1726 ms] 
[2020-02-18T22:14:31.498Z] [PID 3960] [1726 ms] Exit code 2
[2020-02-18T22:14:31.498Z] [PID 3960] [1726 ms] Start: Run in container: test -d /root/.vscode-server/bin/c47d83b293181d9be64f27ff093689e8e7aed054
[2020-02-18T22:14:31.499Z] [PID 3960] [1727 ms] 
[2020-02-18T22:14:31.501Z] [PID 3960] [1729 ms] 
[2020-02-18T22:14:31.501Z] [PID 3960] [1729 ms] Start: Launching Remote-Containers helper.
[2020-02-18T22:14:31.502Z] [PID 3960] [1730 ms] Start: Run in container: # Copy .gitconfig
[2020-02-18T22:14:31.522Z] [PID 3960] [1750 ms] .gitconfig exists
[2020-02-18T22:14:31.522Z] [PID 3960] [1750 ms] 
[2020-02-18T22:14:31.522Z] [PID 3960] [1750 ms] Start: Run in container: command -v git >/dev/null 2>&1 && git config --global credential.helper '!f() { node /tmp/vscode-remote-containers-a4ab1c8c63ca57488f8d83493ca2d9ffee19adcb.js $*; }; f' || true
[2020-02-18T22:14:31.551Z] [PID 3960] [1779 ms] 
[2020-02-18T22:14:31.551Z] [PID 3960] [1779 ms] 
[2020-02-18T22:14:31.551Z] [PID 3960] [1779 ms] Start: Run in container: cat <<'EOF-/tmp/vscode-remote-containers-a4ab1c8c63ca57488f8d83493ca2d9ffee19adcb.js' >/tmp/vscode-remote-containers-a4ab1c8c63ca57488f8d83493ca2d9ffee19adcb.js
[2020-02-18T22:14:31.559Z] [PID 3960] [1787 ms] 
[2020-02-18T22:14:31.559Z] [PID 3960] [1787 ms] 
[2020-02-18T22:14:31.559Z] [PID 3960] [1787 ms] Start: Run in container: set -o noclobber ; mkdir -p '/root/.vscode-server/data/Machine' && { > '/root/.vscode-server/data/Machine/.installExtensionsMarker' ; } 2> /dev/null
[2020-02-18T22:14:31.565Z] [PID 3960] [1793 ms] 
[2020-02-18T22:14:31.565Z] [PID 3960] [1793 ms] 
[2020-02-18T22:14:31.565Z] [PID 3960] [1793 ms] Exit code 2
[2020-02-18T22:14:31.565Z] [PID 3960] [1793 ms] Start: Run in container: cat <<'EOF-/tmp/vscode-remote-containers-server-a4ab1c8c63ca57488f8d83493ca2d9ffee19adcb.js' >/tmp/vscode-remote-containers-server-a4ab1c8c63ca57488f8d83493ca2d9ffee19adcb.js
[2020-02-18T22:14:31.573Z] [PID 3960] [1801 ms] 
[2020-02-18T22:14:31.573Z] [PID 3960] [1801 ms] 
[2020-02-18T22:14:31.573Z] [PID 3960] [1801 ms] Start: Run in container: set -o noclobber ; mkdir -p '/root/.vscode-server/data/Machine' && { > '/root/.vscode-server/data/Machine/.postCreateCommandMarker' ; } 2> /dev/null
[2020-02-18T22:14:31.594Z] [PID 3960] [1822 ms] 
[2020-02-18T22:14:31.594Z] [PID 3960] [1822 ms] 
[2020-02-18T22:14:31.596Z] [PID 3960] [1824 ms] Start: Run in container: test -f /root/.vscode-server/.postCreateCommandMarker
[2020-02-18T22:14:31.735Z] [PID 3960] [1963 ms] 
[2020-02-18T22:14:31.735Z] [PID 3960] [1963 ms] 
[2020-02-18T22:14:31.735Z] [PID 3960] [1963 ms] Exit code 1
[2020-02-18T22:14:31.736Z] [PID 3960] [1964 ms] Start: Run in container: for pid in `cd /proc && ls -d [0-9]*`; do { echo $pid ; readlink -f /proc/$pid/cwd ; xargs -0 < /proc/$pid/environ ; xargs -0 < /proc/$pid/cmdline ; } ; echo ; done 2>/dev/null
[2020-02-18T22:14:31.936Z] [PID 3960] [2164 ms] Extension host agent is already running.
[2020-02-18T22:14:31.936Z] [PID 3960] [2164 ms] Start: Run in container: cat /root/.vscode-server/bin/c47d83b293181d9be64f27ff093689e8e7aed054/.devport 2>/dev/null
[2020-02-18T22:14:31.950Z] [PID 3960] [2178 ms] 44539
[2020-02-18T22:14:31.951Z] [PID 3960] [2179 ms] 
[2020-02-18T22:14:31.954Z] [PID 3960] [2182 ms] Forwarding local port 44539 to container port 44539
[2020-02-18T22:14:31.968Z] [PID 3960] [2196 ms] 

------------------------
Published Ports:
8000/tcp -> 0.0.0.0:8002
------------------------

[2020-02-18T22:14:39.444Z] [PID 2440] [0 ms] Start: Resolving remote
[2020-02-18T22:14:39.446Z] [PID 2440] [2 ms] Setting up container for folder or workspace: c:\Git\pawn


[2020-02-18T22:14:39.467Z] [PID 2440] [23 ms] Start: Check Docker is running
[2020-02-18T22:14:40.810Z] [PID 2440] [1366 ms] Start: Run: docker-compose -f c:\Git\pawn\docker-compose_dev.yml -f c:\Git\pawn\.devcontainer\docker-compose.yml config --services
[2020-02-18T22:14:43.581Z] [PID 2440] [4137 ms] Start: Run: docker-compose --project-name pawn -f c:\Git\pawn\docker-compose_dev.yml -f c:\Git\pawn\.devcontainer\docker-compose.yml up -d --build
[2020-02-18T22:14:43.987Z] [PID 2440] [4543 ms] 
[2020-02-18T22:14:45.145Z] [PID 2440] [5701 ms] Building pawn
[2020-02-18T22:14:46.409Z] [PID 2440] [6965 ms] Step 1/11 : FROM tiangolo/uwsgi-nginx:python3.7
 ---> f9f266fad8c9
Step 2/11 : ENV PYTHONUNBUFFERED=1
 ---> Using cache
 ---> 0cb5c2150773
Step 3/11 : RUN apt-get update &&     apt-get install -y nano &&     apt-get install -y fish
 ---> Using cache
 ---> 1f493147f513
Step 4/11 : WORKDIR /src
 ---> Using cache
 ---> e02f607a2427
Step 5/11 : COPY /requirements.txt /requirements_dev.txt /src/
 ---> Using cache
 ---> 700506b32671
Step 6/11 : RUN pip install -r requirements_dev.txt
 ---> Using cache
 ---> 99c42835468c
Step 7/11 : COPY ./ /src/
[2020-02-18T22:14:46.880Z] [PID 2440] [7436 ms]  ---> 18d61ea27e48
Step 8/11 : EXPOSE 8000
[2020-02-18T22:14:46.923Z] [PID 2440] [7479 ms]  ---> Running in 4d1725d080ad
[2020-02-18T22:14:47.033Z] [PID 2440] [7589 ms] Removing intermediate container 4d1725d080ad
 ---> 546ea38741be
Step 9/11 : STOPSIGNAL SIGINT
[2020-02-18T22:14:47.111Z] [PID 2440] [7667 ms]  ---> Running in 2eab1620b673
[2020-02-18T22:14:47.218Z] [PID 2440] [7774 ms] Removing intermediate container 2eab1620b673
 ---> 4d21b8ee58c4
Step 10/11 : ENTRYPOINT ["python", "manage.py"]
[2020-02-18T22:14:47.268Z] [PID 2440] [7824 ms]  ---> Running in 746cdbf30ffd
[2020-02-18T22:14:47.323Z] [PID 2440] [7879 ms] 
[2020-02-18T22:14:47.403Z] [PID 2440] [7959 ms] Removing intermediate container 746cdbf30ffd
 ---> b8bc3720789e

[2020-02-18T22:14:47.437Z] [PID 2440] [7993 ms] 

Step 11/11 : CMD ["runserver_plus", "--nopin", "0.0.0.0:8000"]
[2020-02-18T22:14:47.470Z] [PID 2440] [8026 ms]  ---> Running in 6c6eb5426c0b
[2020-02-18T22:14:47.594Z] [PID 2440] [8150 ms] Removing intermediate container 6c6eb5426c0b
 ---> 7a1835887065

[2020-02-18T22:14:47.626Z] [PID 2440] [8182 ms] 


Successfully built 7a1835887065
Successfully tagged pawn_pawn:latest
[2020-02-18T22:14:47.663Z] [PID 2440] [8219 ms] Creating pawn ...
[2020-02-18T22:14:48.191Z] [PID 2440] [8747 ms] 

[2020-02-18T22:14:48.235Z] [PID 2440] [8791 ms] 

[2020-02-18T22:17:27.725Z] [PID 19128] [1 ms] Start: Resolving remote
[2020-02-18T22:17:27.729Z] [PID 19128] [5 ms] Setting up container for folder or workspace: c:\Git\pawn


[2020-02-18T22:17:27.751Z] [PID 19128] [27 ms] Start: Check Docker is running
[2020-02-18T22:17:29.013Z] [PID 19128] [1289 ms] Start: Run: docker-compose -f c:\Git\pawn\docker-compose_dev.yml -f c:\Git\pawn\.devcontainer\docker-compose.yml config --services
[2020-02-18T22:17:30.451Z] [PID 19128] [2727 ms] Start: Run: docker-compose --project-name pawn -f c:\Git\pawn\docker-compose_dev.yml -f c:\Git\pawn\.devcontainer\docker-compose.yml up -d --build
[2020-02-18T22:17:30.837Z] [PID 19128] [3113 ms] 
[2020-02-18T22:17:32.142Z] [PID 19128] [4418 ms] Creating network "pawn_default" with the default driver
[2020-02-18T22:17:32.346Z] [PID 19128] [4622 ms] Building pawn
[2020-02-18T22:17:33.108Z] [PID 19128] [5384 ms] Step 1/11 : FROM tiangolo/uwsgi-nginx:python3.7
 ---> f9f266fad8c9
Step 2/11 : ENV PYTHONUNBUFFERED=1
 ---> Using cache
 ---> 0cb5c2150773
Step 3/11 : RUN apt-get update &&     apt-get install -y nano &&     apt-get install -y fish
 ---> Using cache
 ---> 1f493147f513
Step 4/11 : WORKDIR /src
 ---> Using cache
 ---> e02f607a2427
Step 5/11 : COPY /requirements.txt /requirements_dev.txt /src/
 ---> Using cache
 ---> 700506b32671
Step 6/11 : RUN pip install -r requirements_dev.txt
 ---> Using cache
 ---> 99c42835468c
Step 7/11 : COPY ./ /src/
[2020-02-18T22:17:33.633Z] [PID 19128] [5909 ms]  ---> 3c89c4996a1f
Step 8/11 : EXPOSE 8000
[2020-02-18T22:17:33.705Z] [PID 19128] [5981 ms]  ---> Running in efcb71f276ef
[2020-02-18T22:17:33.870Z] [PID 19128] [6146 ms] Removing intermediate container efcb71f276ef
 ---> 31c628636b51
Step 9/11 : STOPSIGNAL SIGINT

[2020-02-18T22:17:33.906Z] [PID 19128] [6182 ms] 

[2020-02-18T22:17:33.948Z] [PID 19128] [6224 ms]  ---> Running in e60edadcab85
[2020-02-18T22:17:34.069Z] [PID 19128] [6345 ms] Removing intermediate container e60edadcab85
 ---> 7a170d7e8963
Step 10/11 : ENTRYPOINT ["python", "manage.py"]
[2020-02-18T22:17:34.115Z] [PID 19128] [6391 ms]  ---> Running in f010dd9960b1
[2020-02-18T22:17:34.257Z] [PID 19128] [6533 ms] Removing intermediate container f010dd9960b1
 ---> a76bd8b91abb
Step 11/11 : CMD ["runserver_plus", "--nopin", "0.0.0.0:8000"]
[2020-02-18T22:17:34.305Z] [PID 19128] [6581 ms]  ---> Running in 7921280e0656
[2020-02-18T22:17:34.420Z] [PID 19128] [6696 ms] Removing intermediate container 7921280e0656
 ---> 656c0654011e

[2020-02-18T22:17:34.473Z] [PID 19128] [6749 ms] Successfully built 656c0654011e
Successfully tagged pawn_pawn:latest
[2020-02-18T22:17:34.523Z] [PID 19128] [6799 ms] Creating pawn ...
[2020-02-18T22:17:35.054Z] [PID 19128] [7330 ms] 
Creating pawn ... done


[2020-02-18T22:17:35.171Z] [PID 19128] [7447 ms] 
[2020-02-18T22:17:35.274Z] [PID 19128] [7550 ms] Start: Inspecting container
[2020-02-18T22:17:35.307Z] [PID 19128] [7583 ms] Start: Run in container: uname -m
[2020-02-18T22:17:35.564Z] [PID 19128] [7840 ms] x86_64
[2020-02-18T22:17:35.565Z] [PID 19128] [7841 ms] 
[2020-02-18T22:17:35.565Z] [PID 19128] [7841 ms] Start: Run in container: (cat /etc/os-release || cat /usr/lib/os-release) 2>/dev/null
[2020-02-18T22:17:35.569Z] [PID 19128] [7845 ms] PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
VERSION_CODENAME=stretch
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
[2020-02-18T22:17:35.570Z] [PID 19128] [7846 ms] 
[2020-02-18T22:17:35.570Z] [PID 19128] [7846 ms] Start: Run in container: cat /etc/passwd
[2020-02-18T22:17:35.574Z] [PID 19128] [7850 ms] Start: Updating configuration state
[2020-02-18T22:17:35.606Z] [PID 19128] [7882 ms] Start: Setup shutdown monitor
[2020-02-18T22:17:35.611Z] [PID 19128] [7887 ms] Forking shutdown monitor: c:\Users\danie\.vscode\extensions\ms-vscode-remote.remote-containers-0.101.0\dist\shutdownMonitorProcess \\.\pipe\vscode-remote-containers-2f561f66c64a3539638e5eaee1f675451cd16ff6-sock dockerCompose Info c:\Users\danie\AppData\Roaming\Code\logs\20200218T230738\exthost1\ms-vscode-remote.remote-containers
[2020-02-18T22:17:35.619Z] [PID 19128] [7895 ms] Start: Run in container: test -d /root/.vscode-server
[2020-02-18T22:17:35.649Z] [PID 19128] [7925 ms] 
[2020-02-18T22:17:35.649Z] [PID 19128] [7925 ms] 
[2020-02-18T22:17:35.649Z] [PID 19128] [7925 ms] Exit code 1
[2020-02-18T22:17:35.649Z] [PID 19128] [7925 ms] Start: Run in container: test -d /root/.vscode-remote
[2020-02-18T22:17:35.654Z] [PID 19128] [7930 ms] 
[2020-02-18T22:17:35.655Z] [PID 19128] [7931 ms] 
[2020-02-18T22:17:35.655Z] [PID 19128] [7931 ms] Exit code 1
[2020-02-18T22:17:35.655Z] [PID 19128] [7931 ms] Start: Run in container: set -o noclobber ; mkdir -p '/root/.vscode-server/data/Machine' && { > '/root/.vscode-server/data/Machine/.writeMachineSettingsMarker' ; } 2> /dev/null
[2020-02-18T22:17:35.665Z] [PID 19128] [7941 ms] 
[2020-02-18T22:17:35.665Z] [PID 19128] [7941 ms] 
[2020-02-18T22:17:35.666Z] [PID 19128] [7942 ms] Start: Run in container: mkdir -p '/root/.vscode-server/data/Machine' && cat >'/root/.vscode-server/data/Machine/settings.json' <<'settingsJSON'
[2020-02-18T22:17:35.669Z] [PID 19128] [7945 ms] 
[2020-02-18T22:17:35.669Z] [PID 19128] [7945 ms] 
[2020-02-18T22:17:35.670Z] [PID 19128] [7946 ms] Start: Run in container: test -d /root/.vscode-server/bin/c47d83b293181d9be64f27ff093689e8e7aed054
[2020-02-18T22:17:35.672Z] [PID 19128] [7948 ms] 
[2020-02-18T22:17:35.672Z] [PID 19128] [7948 ms] 
[2020-02-18T22:17:35.672Z] [PID 19128] [7948 ms] Exit code 1
[2020-02-18T22:17:35.672Z] [PID 19128] [7948 ms] Installing VS Code Server for commit c47d83b293181d9be64f27ff093689e8e7aed054
[2020-02-18T22:17:35.672Z] [PID 19128] [7948 ms] Start: Run in container: mkdir -p /root/.vscode-server/bin/c47d83b293181d9be64f27ff093689e8e7aed054_1582064255672
[2020-02-18T22:17:35.675Z] [PID 19128] [7951 ms] 
[2020-02-18T22:17:35.675Z] [PID 19128] [7951 ms] 
[2020-02-18T22:17:35.677Z] [PID 19128] [7953 ms] Start: Run: docker exec -w /root/.vscode-server/bin/c47d83b293181d9be64f27ff093689e8e7aed054_1582064255672 -u root -i be183cefa591d5d82404447a2ad0ef679baf13bd08d98810d3658f4572816733 tar --no-same-owner -xz --strip-components 1
[2020-02-18T22:17:36.443Z] [PID 19128] [8719 ms] 
[2020-02-18T22:17:36.444Z] [PID 19128] [8720 ms] 

[2020-02-18T22:17:36.446Z] [PID 19128] [8722 ms] Command failed: C:\Program Files\Docker\Docker\resources\bin\docker.exe exec -w /root/.vscode-server/bin/c47d83b293181d9be64f27ff093689e8e7aed054_1582064255672 -u root -i be183cefa591d5d82404447a2ad0ef679baf13bd08d98810d3658f4572816733 tar --no-same-owner -xz --strip-components 1

```
Maybe you see something in there.

And yes, I'm able to `docker-compose up` and `Remote-Containers: Attach to Running Container`. So I really don't know what's going on that I'm able to make a devcontainer from the Dockerfile, and am able to attach to a running container, but can't make a working devcontainer from the compose file.

**Comment by chrmarti** [comment]
I would expect some additional error output from the failing command itself, that seems to be lost somehow. Could you check if the container is still running at that point?

**Comment by danielfiniki** [comment]
The container stops as soon as I get the error message. Did you try to start a similar project on your machine? I'd think it should mostly depend on the base image you're using, but then making a devcontainer from the Dockerfile shouldn't work either.

**Comment by chrmarti** [comment]
The container needs to keep running on its own. I assume
```
ENTRYPOINT ["python", "manage.py"]
CMD ["runserver_plus", "--nopin", "0.0.0.0:8000"]
```
keeps it running?

Could you also append your devcontainer.json?

**Comment by danielfiniki** [comment]
Yes, those two lines start the development server and keep it running.

The generated devcontainer.json is this one:
```
// For format details, see https://aka.ms/vscode-remote/devcontainer.json or this file's README at:
// https://github.com/microsoft/vscode-dev-containers/tree/v0.101.1/containers/docker-existing-docker-compose
// If you want to run as a non-root user in the container, see .devcontainer/docker-compose.yml.
{
	"name": "Existing Docker Compose (Extend)",

	// Update the 'dockerComposeFile' list if you have more compose files or use different names.
	// The .devcontainer/docker-compose.yml file contains any overrides you need/want to make.
	"dockerComposeFile": [
		"..\\docker-compose_dev.yml",
		"docker-compose.yml"
	],

	// The 'service' property is the name of the service for the container that VS Code should
	// use. Update this value and .devcontainer/docker-compose.yml to the real service name.
	"service": "pawn",

	// The optional 'workspaceFolder' property is the path VS Code should open by default when
	// connected. This is typically a file mount in .devcontainer/docker-compose.yml
	"workspaceFolder": "/workspace",

	// Set *default* container specific settings.json values on container create.
	"settings": { 
		"terminal.integrated.shell.linux": null
	},

	// Add the IDs of extensions you want installed when the container is created.
	"extensions": []

	// Uncomment the next line if you want start specific services in your Docker Compose config.
	// "runServices": [],

	// Uncomment the next line if you want to keep your containers running after VS Code shuts down.
	// "shutdownAction": "none",

	// Uncomment the next line to run commands after the container is created - for example installing git.
	// "postCreateCommand": "apt-get update && apt-get install -y git",

	// Uncomment to connect as a non-root user. See https://aka.ms/vscode-remote/containers/non-root.
	// "remoteUser": "vscode"
}
```

And the generated docker-compose.yml file:
```
// For format details, see https://aka.ms/vscode-remote/devcontainer.json or this file's README at:
// https://github.com/microsoft/vscode-dev-containers/tree/v0.101.1/containers/docker-existing-docker-compose
// If you want to run as a non-root user in the container, see .devcontainer/docker-compose.yml.
{
	"name": "Existing Docker Compose (Extend)",

	// Update the 'dockerComposeFile' list if you have more compose files or use different names.
	// The .devcontainer/docker-compose.yml file contains any overrides you need/want to make.
	"dockerComposeFile": [
		"..\\docker-compose_dev.yml",
		"docker-compose.yml"
	],

	// The 'service' property is the name of the service for the container that VS Code should
	// use. Update this value and .devcontainer/docker-compose.yml to the real service name.
	"service": "pawn",

	// The optional 'workspaceFolder' property is the path VS Code should open by default when
	// connected. This is typically a file mount in .devcontainer/docker-compose.yml
	"workspaceFolder": "/workspace",

	// Set *default* container specific settings.json values on container create.
	"settings": { 
		"terminal.integrated.shell.linux": null
	},

	// Add the IDs of extensions you want installed when the container is created.
	"extensions": []

	// Uncomment the next line if you want start specific services in your Docker Compose config.
	// "runServices": [],

	// Uncomment the next line if you want to keep your containers running after VS Code shuts down.
	// "shutdownAction": "none",

	// Uncomment the next line to run commands after the container is created - for example installing git.
	// "postCreateCommand": "apt-get update && apt-get install -y git",

	// Uncomment to connect as a non-root user. See https://aka.ms/vscode-remote/containers/non-root.
	// "remoteUser": "vscode"
}
```

**Comment by airtonix** [comment]
My error here gets a bit further: 

```
[2020-04-22T23:00:28.080Z] [PID 74448] [18351 ms] 
[2020-04-22T23:00:28.080Z] [PID 74448] [18351 ms] Exit code 1
[2020-04-22T23:00:28.080Z] [PID 74448] [18351 ms] Installing VS Code Server for commit ff915844119ce9485abfe8aa9076ec76b5300ddd
[2020-04-22T23:00:28.081Z] [PID 74448] [18352 ms] Start: Run in container: mkdir -p /root/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec76b5300ddd_1587596428081
[2020-04-22T23:00:28.093Z] [PID 74448] [18364 ms] 
[2020-04-22T23:00:28.093Z] [PID 74448] [18364 ms] 
[2020-04-22T23:00:28.094Z] [PID 74448] [18365 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec76b5300ddd_1587596428081 fedf1ef2e7fab6d908dd7f9e2b3ad261f512327861a2d91ef22bca7dc4414819 tar --no-same-owner -xz --strip-components 1
[2020-04-22T23:00:28.256Z] [PID 74448] [18527 ms] 
[2020-04-22T23:00:28.256Z] [PID 74448] [18527 ms] unknown shorthand flag: 'w' in -w
See 'docker exec --help'.
```

**Comment by chrmarti** [comment]
`-w` for `exec` is only available from API 1.35+. Cannot find the corresponding server version at the moment.
